src/main/java/uk/gov/companieshouse/api/testdata/service/impl/FilingHistoryServiceImpl.java

```diff
@@ -215,22 +213,36 @@ private List<AssociatedFiling> createAssociatedFilings(Instant dayTimeNow, Insta
         return associatedFilings;
     }
 
-    private Links createLinks(String fhType, String companyNumber, String id) {
+    private Links createLinks(String companyNumber, String id, boolean includeMetaData) {
         Links links = new Links();
         links.setSelf("/company/" + companyNumber + "/filing-history/" + id);
-        if ("AP01".equals(fhType) || TYPE.equals(fhType)) {
+        if (includeMetaData) {
             links.setDocumentMetadata("document/" + id);
         }
         return links;
     }
 
-    private DescriptionValues createDescriptionValues(String type , Instant dayNow) {
+    DescriptionValues createDescriptionValues(String type, Instant dayNow, FilingHistorySpec fhspec) {
         var descriptionValues = new DescriptionValues();
-        if("AP01".equals(type)) {
-            descriptionValues.setAppointmentDate(dayNow);
-            descriptionValues.setOfficerName("Mr John Test");
-        } else {
-            descriptionValues.setChargeNumber(String.valueOf(randomService.getNumber(12)));
+        LOG.debug("TYPE " + type);
+        switch (type) {
+            case "AP01" -> {
+                descriptionValues.setAppointmentDate(dayNow);
+                descriptionValues.setOfficerName("Mr John Test");
+            }
+            case "SH01" -> {
+                List<Capital> capitalList = new ArrayList<>();
```

Comment:
This is a micro-optimisation so feel free to ignore is but as you already know the size of the list you can do:

```suggestion
                List<Capital> capitalList = new ArrayList<>(fhspec.getDescriptionValues().getCapital().size());
```

As in this case you're using `fhspec.getDescriptionValues().getCapital()` twice it might be best to move that to another variable though.

Reply:
updated the script